ZEP-XXX-journal-record-format.md

```diff
@@ -0,0 +1,239 @@
+---
+title: Journal record format
+authors:
+  - Deepthi Akkoorath
+reviewers:
+  - TBD
+approvers:
+  - TBD
+editor: TBD
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+
+---
+
+# Summary
+[summary]: #summary
+
+
+This document describes the format of a journal record as defined in the upcoming release Zeebe 1.0.0.
+
+In this document we describe fields added by the raft and the journal to the record.
+The fields added by the process engine is out of scope for this document.
+
+Different fields of a journal record is encoded using [SBE](https://github.com/real-logic/simple-binary-encoding). Please see Zeebe repository to find the SBE schema.
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+
+## Journal layout
+The journal contains one or more segments. Each segment is a file that contains a sequence of records ordered by index.
+
+
+TODO: JournalSegmentDescriptor - first 64 bytes of a segment
+
```

Comment:
I would prefer it to document journal record format. But since we discussed about documenting the whole journal, I added a small section here for the descriptor. If we are planning to have another document, then I would say only record format here.